File: ProjectEuler/001-050/S40.cs
using System.Text;

namespace ProjectEuler
{
    public class S40 : ISolution
    {
        public string GetAnswer()
        {
            StringBuilder sb = new StringBuilder(".");
            for (int i = 1; ; i++)
            {
                sb.Append(i.ToString());
                if (sb.Length > 1000000)
                {
                    break;
                }
            }

            string s = sb.ToString();
            int answer = (s[1] - '0') * (s[10] - '0') * (s[100] - '0') *
                (s[1000] - '0') * (s[10000] - '0') * (s[100000] - '0') * (s[1000000] - '0');

            return answer.ToString();
        }
    }
}

File: src/001-050/S40.cc
#include "S40.h"
#include "../Utils.h"
#include <string>

using namespace std;

namespace ProjectEuler
{
std::string S40::GetAnswer()
{
	std::string sb = ".";
	for (int i = 1;; i++)
	{
		sb += std::to_string(i);
		if (sb.size() > 1000000)
		{
			break;
		}
	}

	int answer = (sb[1] - '0') * (sb[10] - '0') * (sb[100] - '0') * (sb[1000] - '0') * (sb[10000] - '0') *
				 (sb[100000] - '0') * (sb[1000000] - '0');

	return std::to_string(answer);
}

} // namespace ProjectEuler
